include/veriblock/alt-util.hpp

```diff
@@ -15,12 +15,12 @@
 namespace altintegration {
 
 template <typename BlockTree>
-std::vector<std::vector<uint8_t>> getLastKnownBlocks(const BlockTree& tree,
-                                                     size_t size) {
+std::vector<std::vector<uint8_t>> getLastKnownBlocks(
+    std::shared_ptr<BlockTree> tree, size_t size) {
```

Comment:
revert back. smart pointer is not needed here, as you do not pass ownership to this function